src/instructions/interrupts.rs

```diff
@@ -16,10 +15,11 @@ pub fn are_enabled() -> bool {
 /// This is a wrapper around the `sti` instruction.
 #[inline]
 pub fn enable() {
-    // Prevent earlier writes to be moved beyond this point
-    compiler_fence(Ordering::Release);
+    // Omit `nomem` to imitate a lock release.
+    // Otherwise, the compiler is free to move
+    // reads and writes through this asm block.
```

Comment:
```suggestion
    // Omit `nomem` to imitate a lock release. Otherwise, the compiler
    // is free to move reads and writes through this asm block.
```